packages/pieces/date-helper/src/lib/actions/next-day-of-week.ts

```diff
@@ -0,0 +1,97 @@
+			displayName : 'Weekday',
+			description : "The weekday that you would like to get the date and time of.",
+            options     : {
+                options: [
+                    { label: 'Sunday',    value: 0 },
+                    { label: 'Monday',    value: 1 },
+                    { label: 'Tuesday',   value: 2 },
+                    { label: 'Wednesday', value: 3 },
+                    { label: 'Thursday',  value: 4 },
+                    { label: 'Friday',    value: 5 },
+                    { label: 'Saturday',  value: 6 },
+                ],
+            },
+            required     : true,
+        }),
+        time: Property.ShortText({
+            displayName  : '24h Time',
+            description  : "The time that you would like to get the date and time of. This must be in 24h format.",
+            required     : false,
+            defaultValue : "00:00",
+            validators   : [ Validators.pattern(/^\d\d:\d\d$/) ],
+		}),
+        currentTime: Property.Checkbox({
+			displayName  : 'Use Current Time',
+			description  : 'If checked, the current time will be used instead of the time specified above.',
+			required     : false,
+			defaultValue : false
+		}),
+        timeFormat: Property.StaticDropdown({
+            displayName : 'To Time Format',
+            description : timeFormatDescription,
+            options     : {
+                options: optionalTimeFormats,
+            },
+            required     : true,
+            defaultValue : timeFormat.format00,
+        }),
+        timeZone: Property.StaticDropdown<string>({
+            displayName : 'Time Zone',
+            options     : {
+                options : timeZoneOptions,
+            },
+            required     : true,
+            defaultValue : 'UTC',
+        }),
+	},
+	async run(context) {
+        const timeFormat  = context.propsValue.timeFormat;
+        const timeZone    = context.propsValue.timeZone as string;
+        const dayIndex    = context.propsValue.weekday as number;
+        const currentTime = context.propsValue.currentTime as boolean;
+        let   time        = context.propsValue.time as string;
+
+        const now = new Date();
+
+        if (currentTime === true) {
+            time = `${now.getHours()}:${now.getMinutes()}`;
+        }
+
+        const [hours, minutes] = time.split(':').map(Number);
+
+        // Validate inputs
+        if (dayIndex < 0 || dayIndex > 6 || hours < 0 || hours > 23 || minutes < 0 || minutes > 59) {
+            throw new Error(`Invalid input \ndayIndex: ${dayIndex} \nhours: ${hours} \nminutes: ${minutes}`);
+        }
+
+        if (typeof timeFormat !== 'string') {
+            throw new Error(`Output format is not a string \noutput format: ${JSON.stringify(timeFormat)}`);
+        }
+
+        // Calculate the day difference
+        let dayDiff = dayIndex - now.getDay();
+        if (dayDiff < 0 || (dayDiff === 0 && now.getTime() > new Date().getTime())) {
```

Comment:
now.getTime() > new Date().getTime() basically always returns false, since now is initiated as new Date() in line 63 and then goes into the condition without taking into account the time the user has entered in the time prop

Reply:
I noticed this this morning. All fixed now.